cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2660,60 +2616,97 @@ int getPodIndexFromPvcName(String pvcName)  {
 
         @SuppressWarnings("deprecation")
         private final Future<ReconciliationState> getEntityOperatorDescription() {
```

Comment:
I wonder if this could be somehow split into 3 small methods: `getEntityoperatorDescription`, `getEntityTopicOperatorDescription` and `getEntityUserOperatorDescription` which would split the work and make it easier to chain the async work. But that is just a quick idea, nto sure how well would it work. Right now it seems to dupliacate a bit on the variants with both operators, just one of them etc.